packages/atlas/src/components/Text/Text.styles.ts

```diff
@@ -3,127 +3,146 @@ import styled from '@emotion/styled'
 
 import { cVar } from '@/styles'
 
-type TextProps = {
+export type TextBaseProps = {
   color?: string
-  isSecondary?: boolean
+  secondary?: boolean
   clampAfterLine?: number
+  margin?: MarginProps
 }
 
-const clampStyles = ({ clampAfterLine }: TextProps) => css`
+type MarginProps =
+  | {
+      top?: string
+      bottom?: string
+      left?: string
+      right?: string
+    }
+  | string
+
+const marginStyles = ({ margin }: TextBaseProps) =>
+  typeof margin !== 'string' && !!margin
+    ? css`
+        margin: ${margin.top} ${margin.right} ${margin.bottom} ${margin.left};
+      `
+    : css`
+        margin: ${margin ?? 0};
```

Comment:
I think this is conflicting with similar functionality Rafał is introducing in his PR: https://github.com/Joystream/atlas/pull/2218/files#diff-f4908e2ae287f2ff9c86b053a80805d5fc9ca31f1ff659566496391e1020cb0fR30

@eldiegod @rafalpawlow Please sync up on that so we use the same approach in both PRs